gson/src/test/java/com/google/gson/stream/JsonReaderTest.java

```diff
@@ -38,6 +38,103 @@
 
 @SuppressWarnings("resource")
 public final class JsonReaderTest {
+  @Test
+  public void testEscapedNewlineNotAllowedInStrictMode() throws IOException {
+    String json = "\"\\\n\"";
+    JsonReader reader = new JsonReader(reader(json));
+    reader.setStrict(true);
+    try {
+      reader.nextString();
+      fail();
+    } catch (IOException expected) {
+
```

Comment:
If you want to leave an empty catch block you shoud put a comment inside. Sometime like:

```
// OK: reason why the block is empty
```

Otherwise you can check if the exception has the correct message:
```java
assertThat(e).hasMessageThat().isEqualTo("...");
```

---

Here some exemple:

https://github.com/google/gson/blob/742fd50ff74350462d8dcf80941efce219e187dd/gson/src/test/java/com/google/gson/JsonPrimitiveTest.java#L113-L115

https://github.com/google/gson/blob/742fd50ff74350462d8dcf80941efce219e187dd/gson/src/main/java/com/google/gson/internal/ReflectionAccessFilterHelper.java#L100-L102

Reply:
I think we can use `assertThrows` here, and yes, it would be good to check the exception message.

Reply:
Sorry, I forgot we're compiling tests in Java 7 mode, so lambdas aren't allowed and `assertThrows` is inconvenient. I think we could actually compile the tests in Java 11 mode, with Gson itself still being Java 7. But that's a different project.

Reply:
Now that this PR #2324 has been merged the tests are compiled with `JDK-11`. So, you can use lambda (in tests). 😄

Reply:
I think we now assert the commit message, without using a lambda.